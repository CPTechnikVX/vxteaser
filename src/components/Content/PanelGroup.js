import React     from 'react';
import Constants from '../../utils/Constants';
import PropTypes from 'prop-types';

/**
 * Standard group of panels
 */
export default class PanelGroup extends React.PureComponent {
	static propTypes = {
		/** @ignore */
		children: PropTypes.node,
	};

	render() {
		const {children} = this.props;

		return <div className={Constants.ClassName.PanelGroup}>{children}</div>;
	}
}
