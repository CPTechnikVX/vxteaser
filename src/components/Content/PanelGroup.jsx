import React      from 'react';
import Constants  from '../../utils/Constants';
import PropTypes  from 'prop-types';
import classnames from 'classnames';

/**
 * Standard group of panels
 */
export default class PanelGroup extends React.PureComponent {
	render() {
		const {children, config} = this.props;
		const classList          = [Constants.ClassName.PanelGroup];

		if (config && typeof config.aspectRatio !== 'undefined') {
			classList.push(Constants.ClassName.PanelGroup + '--' + Constants.Element.Tile);
		}

		return <div className={classnames(classList)}>{children}</div>;
	}
}

PanelGroup.propTypes = {
	/** @ignore */
	children: PropTypes.node,
	/** @ignore */
	config:   PropTypes.object,
};
