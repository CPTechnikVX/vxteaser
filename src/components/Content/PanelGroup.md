```html
<PanelGroup>
	<Panel>
	  .... some content hier
	</Panel>
</PanelGroup>
```
