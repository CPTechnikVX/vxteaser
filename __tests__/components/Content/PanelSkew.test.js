import React      from 'react';
import {render}   from '@testing-library/react';
import PanelGroup from '../../../src/components/Content/PanelGroup';

describe('render panel group', () => {
    test('with content', () => {
        const {container} = render(
                <PanelGroup>Content</PanelGroup>
        );
        expect(container.firstChild).toMatchSnapshot();
    });
});
