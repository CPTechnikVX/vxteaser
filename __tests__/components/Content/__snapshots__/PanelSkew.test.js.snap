// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`render panel group with content 1`] = `
<div
  class="vxteaser-panel-group"
>
  Content
</div>
`;
